rir/src/compiler/analysis/reference_count.h

```diff
@@ -8,29 +8,92 @@ namespace rir {
 namespace pir {
 
 struct AUses {
+
+    /*
+     * Lattice:
+     *
+     *                    Multiple
+     *                     /      \
+     *   AlreadyIncremented        Once
+     *                     \      /
+     *                       None
+     *
+     * Definitions:
+     *   reuse            :=   override if refcount is == 0
+     */
+
     enum Kind {
-        None = 0,
-        Once = 1,
-        Multiple = 2,
-        Destructive = 3,
+        // 1) Static refcount is safe
+        None,
+        Once,               // At least one use, which reuses input
+        AlreadyIncremented, // Refcount is known to be 1 (ie. used by stvar)
+
+        // 2) Need an ensureNamed
+        Multiple, // At least two uses, at least one (but the last) is reusing
```

Comment:
Nitpick: comment not lined up with the other ones above.

Reply:
this is clang format unfortunately